src/Core/Models/Commands/CommandResult.cs

```diff
@@ -10,3 +12,51 @@ public CommandResult(string error) : this([error]) { }
 
     public CommandResult() : this(Array.Empty<string>()) { }
 }
+
+public class FailureCommandResult : CommandResult
```

Comment:
Currently `CommandResult` has these properties, 
```
    public bool Success => ErrorMessages.Count == 0;
    public bool HasErrors => ErrorMessages.Count > 0;
```
But maybe we can remove them once we have a type for failures. I'm happy to hear any pushback or cases I may have missed.